source/change-streams/tests/README.rst

```diff
@@ -0,0 +1,100 @@
+.. role:: javascript(code)
+  :language: javascript
+
+==================
+Change Streams
+==================
+
+.. contents::
+
+--------
+
+Introduction
+============
+
+The YAML and JSON files in this directory are platform-independent tests that
+drivers can use to prove their conformance to the Change Streams Spec.
+
+Several prose tests, which are not easily expressed in YAML, are also presented
+in this file. Those tests will need to be manually implemented by each driver.
+
+Test Format
+===========
+
+Each YAML files has the following keys:
+
+- ``database_name``: The default database
+- ``collection_name``: The default collection
+- ``defaults``: Default values for each test
+- ``tests``: An array of tests that are to be run independently of each other.
+  Each test will have some of the following fields:
+
+  - ``description``: The name of the test.
+  - ``skip``: Notes that this test is incomplete and should not be run
+  - ``minServerVersion``: The minimum server version to run this test against. If not present, assume there is no minimum server version.
+  - ``maxServerVersion``: Reserved for later use
+  - ``failPoint``: Reserved for later use
+  - ``topology``: The server topology/topologies to run the test against.
+    Valid topologies are ``single``, ``replicaset``, and ``sharded``.
+    Can be either a single topology, or a list of topologies.
+  - ``target``: The entity to run a changeStream on. Valid values are:
+  
+    - ``collection``: Watch changes on collection ``database_name.collection_name``
+    - ``db``: Watch changes on database ``database_name``
+    - ``client``: Watch changes on entire clusters
+
+  - ``numChanges``: The number of changes to expect on a changeStream
+  - ``operations``: Array of documents, each describing an operation. Each document has the following fields:
+
+    - ``database``: Database to run the operation against
+    - ``collection``: Collection to run the operation against
+    - ``commandName``: Name of the command to run
+    - ``arguments``: Array of arguments for the command.
+
+  - ``expectations``: Optional list of command-started events
+  - ``result``: Document with ONE of the following fields:
+
+    - ``error``: Describes an error received during the test
+    - ``success``: An array of documents expected to be received from the changeStream
+
+For every test, if a value does not exist, a value from ``defaults`` should be used in its place.
```

Comment:
This makes it less straightforward to implement a test runner.  Since we have to write ~10 of those, and only write the spec tests once, just repeat in each test, using YAML references where possible to avoid duplication.

Also, none of our other test runners require this so it's out of character to do it here.

Reply:
done